code/modules/mob/living/internals.dm

```diff
@@ -0,0 +1,136 @@
+/mob/living/proc/set_internals(obj/item/tank/source, source_string)
+	return
+
+/mob/living/proc/get_internals()
+	return
+
+// Set internals on or off. Implemented properly on /carbon
+/mob/living/proc/toggle_internals(var/mob/living/user)
+	return
+
+/mob/living/proc/breathing_hole_covered()
+	var/obj/item/mask = get_equipped_item(slot_wear_mask_str)
+	. = (mask && (mask?.item_flags & ITEM_FLAG_AIRTIGHT))
+
+/mob/living/proc/get_possible_internals_sources()
+	. = list("back" = list(get_equipped_item(slot_back_str), "on"))
+
+/mob/living/proc/set_internals_to_best_available_tank(var/breathes_gas = /decl/material/gas/oxygen, var/list/poison_gas = list(/decl/material/gas/chlorine))
+	return
+
+/mob/living/proc/ui_toggle_internals()
+
+	if(incapacitated())
+		return
+
+	if(get_internals())
+		set_internals(null)
+		return
+
+	if(!breathing_hole_covered())
+		to_chat(src, SPAN_WARNING("You are not wearing a suitable mask or helmet."))
+		return
+
+	set_internals_to_best_available_tank()
+
+	if(!get_internals())
+		to_chat(src, SPAN_WARNING("You don't have a tank that is usable as internals."))
+
+/mob/living/toggle_internals(var/mob/living/user)
+
+	var/atom/movable/internal = get_internals()
+	if(internal)
+		visible_message(SPAN_NOTICE("\The [user] disables \the [src]'s internals!"))
+		internal.add_fingerprint(user)
+		set_internals(null)
+		return
+
+	// Check for airtight mask/helmet.
+	var/found_mask = FALSE
+	for(var/slot in global.airtight_slots)
+		var/obj/item/gear = get_equipped_item(slot)
+		if(gear && (gear.item_flags & ITEM_FLAG_AIRTIGHT))
+			found_mask = TRUE
+			break
+
+	if(!found_mask)
+		to_chat(user, SPAN_WARNING("\The [src] does not have a suitable mask or helmet."))
+		return
+
+	// Find an internal source.
+	for(var/slot in global.internal_source_slots)
```

Comment:
Why not check `get_possible_internals_sources()` here and remove `global.internal_source_slots` entirely? Currently this results in a weird disparity between someone turning on their own internals and someone else enabling it for them.